dbus-codegen/src/main.rs

```diff
@@ -47,15 +47,23 @@ Defaults to 'RefClosure'."))
              .help("If present, skips a specific prefix for interface names, e g 'org.freedesktop.DBus.'."))
         .arg(clap::Arg::with_name("futures").short("f").long("futures")
              .help("Generates code to use with futures 0.3 (experimental)"))
+        .arg(clap::Arg::with_name("file").required(false).help("D-Bus XML Introspection file"))
         .get_matches();
 
+	if matches.value_of("destination").is_some() && matches.value_of("file").is_some() {
+		panic!("Expected either xml file path as argument or destination option. But both are provided.");
+	}
+	
     let s = 
     if let Some(dest) = matches.value_of("destination") {
         let path = matches.value_of("path").unwrap_or("/");
         let bus = if matches.is_present("systembus") { dbus::BusType::System } else { dbus::BusType::Session };
         let c = dbus::Connection::get_private(bus).unwrap();
         let p = c.with_path(dest, path, 10000);
         p.introspect().unwrap()
+    } else if let Some(file_path) = matches.value_of("file")  {
+    	let s = std::fs::read_to_string(file_path.to_string()).unwrap();
```

Comment:
```suggestion
    	std::fs::read_to_string(file_path.to_string()).unwrap()
```

Reply:
This doesn't work. A build error is thrown. 
![Screenshot from 2019-04-26 01-18-41](https://user-images.githubusercontent.com/15038177/56764075-81783a80-67c1-11e9-9a22-1dbb3bee10b3.png)